files/views.py

```
@@ -2111,3 +2112,33 @@ def get(self, request, format=None):
         serializer = IndexPageFeaturedSerializer(indexfeatured, many=True, context={"request": request})
         return Response(serializer.data)
+
+
+class MediaKeyView(APIView):
+    """Serve AES-128 decryption key for encrypted HLS streams."""
+
+    def get(self, request, friendly_token):
+        try:
+            media = Media.objects.select_related("user").get(friendly_token=friendly_token)
+        except Media.DoesNotExist:
+            raise Http404
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)